src/encoder.rs

```diff
@@ -350,6 +356,89 @@ impl<T: Pixel> FrameState<T> {
+  ///
+  /// # Panics
+  ///
+  /// Panics if the activity mask is not decimated by 8 on both axes.
+  pub fn compute_activity_mask(&mut self) {
+
+    // Contrast Sensiticity Function
+    let csf: [[f32; 8]; 8] = [
+      [1.608_443, 2.339_554, 2.573_509, 1.608_443, 1.072_295, 0.643_377, 0.504_610, 0.421_887],
+      [2.144_591, 2.144_591, 1.838_221, 1.354_478, 0.989_811, 0.443_708, 0.428_918, 0.467_911],
+      [1.838_221, 1.979_622, 1.608_443, 1.072_295, 0.643_377, 0.451_493, 0.372_972, 0.459_555],
+      [1.838_221, 1.513_829, 1.169_777, 0.887_417, 0.504_610, 0.295_806, 0.321_689, 0.415_082],
+      [1.429_727, 1.169_777, 0.695_543, 0.459_555, 0.378_457, 0.236_102, 0.249_855, 0.334_222],
+      [1.072_295, 0.735_288, 0.467_911, 0.402_111, 0.317_717, 0.247_453, 0.227_744, 0.279_729],
+      [0.525_206, 0.402_111, 0.329_937, 0.295_806, 0.249_855, 0.212_687, 0.214_459, 0.254_803],
+      [0.357_432, 0.279_729, 0.270_896, 0.262_603, 0.229_778, 0.257_351, 0.249_855, 0.259_950]];
+    
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let activity_plane = &mut self.activity_mask;
+    let PlaneConfig { xdec, ydec, .. } = activity_plane.cfg;
+    let activity_mask: &mut [u16] = &mut *activity_plane.data;
+    assert!(xdec == 3 && ydec == 3);
+    let bit_depth = self.bit_depth;
+
+    let aligned_luma = Rect {
+      x: 0_isize,
+      y: 0_isize,
+      width: (width >> xdec) << xdec,
+      height: (height >> ydec) << ydec,
+    };
+    let luma = PlaneRegion::new(luma_plane, aligned_luma);
+
+    for x in 0 .. width >> xdec {
+      for y in 0 .. height >> ydec {
+
+        let block_rect = Area::Rect{
+          x: (x << xdec) as isize,
+          y: (y << ydec) as isize,
+          width: 8,
+          height: 8,
+        };
+        
+        let block = luma.subregion(block_rect);
+        
+        let mut mean: i32 = block.rows_iter().flatten().map(|&val| {let int: i32 = CastFromPrimitive::cast_from(val); int}).sum();
+        mean >>= xdec + ydec;
+        
+        // Calculating deviation in Spatial domain 
+        let deviation: Vec<i16> = block.rows_iter().flatten().map(|&pix| {
+          let pix_int: i16 = CastFromPrimitive::cast_from(pix);
+          match bit_depth {
+            8 => (pix_int - mean as i16) << 1,
+            10 => (pix_int - mean as i16) >> 1,
+            12 => (pix_int - mean as i16) >> 3,
+            _ => panic!("Unsuppported bit-depth"),
+          }
+        }).collect();
+        
+        let tx_type = TxType::DCT_DCT;
+        let tx_size = TxSize::TX_8X8;
+        let mut tx_deviation = [0_i32; 64];
+        forward_transform(&deviation, &mut tx_deviation, tx_size.width(), tx_size, tx_type, 8);
+
+        // Applying CSF to the deviation on tx_domain
+        tx_deviation.iter_mut().zip(csf.iter().flatten()).for_each(|(d, &csf)| *d = (*d as f32 * csf).floor() as i32);
+
+        let mut cs_dev = Plane::<u16>::new(8, 8, 0, 0, 0, 0);
```

Comment:
Note that the pixel values of `cs_dev` are initialized to 128.

Reply:
We may avoid rapidly churning the allocator here by hoisting this `new()` outside the loops. We then need to explicitly fill in each iteration. The following worked for me but is a quick-hack:
```rust
cs_dev.data.copy_from_slice(&[128u16; 128]);
```

Reply:
Given it's a fixed size new, you should be able to replace it with AlignedAlloc instead.

Reply:
Thanks for the hack. It looks cleaner than how I'd usually implement.

Reply:
> ```rust
> cs_dev.data.copy_from_slice(&[128u16; 128]);
> ```

Any reason for the size of the slice being 128? Won't 64 suffice? (For an 8x8 block)

Reply:
It's due to the stride alignment of `PlaneData`, 32 bytes = 16 u16 pixels = 8 cols * 2; an internal detail that leaks here. We may avoid this altogether by accumulating in the frequency domain but then we need to be mindful of scale.

Reply:
> Given it's a fixed size new, you should be able to replace it with AlignedAlloc instead.

AlignedArray instead of Plane::new() for cs_dev?